src/name/verify.rs

```diff
@@ -238,7 +238,10 @@ fn check_presented_id_conforms_to_constraints_in_subtree(
                 Err(Error::NameConstraintViolation)
             }
 
-            _ => Ok(false),
+            _ => {
+                // mismatch between constraint and name types
+                continue;
```

Comment:
I find this code pretty confusing. Should this be returning a `NameIteration::KeepGoing` instead? Why not?

Reply:
will extend the commentary here; this is inside a loop over the constraints. if (for example) the current constraint is on `GeneralName::IpAddress` but the current name is a `GeneralName::DnsName`, we want to check all the other constraints before moving on to the next name.